pkg/providers/instancetype/types.go

```diff
@@ -247,10 +265,16 @@ func habanaGaudis(info *ec2.InstanceTypeInfo) *resource.Quantity {
 	return resources.Quantity(fmt.Sprint(count))
 }
 
-// The number of pods per node is calculated using the formula:
-// max number of ENIs * (IPv4 Addresses per ENI -1) + 2
-// https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt#L20
-func ENILimitedPods(info *ec2.InstanceTypeInfo) *resource.Quantity {
+func ENILimitedPods(info *ec2.InstanceTypeInfo, amiFamily amifamily.AMIFamily) *resource.Quantity {
+	// https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html
+	// The number of pods that you can run per Windows node is equal to the number of IP addresses
+	// available per elastic network interface for the node's instance type, minus one
+	if _, ok := amiFamily.(*amifamily.Windows); ok {
```

Comment:
Ah, apologize. You are correct that we should have this resource capacity in a separate function. This calculation is based off of the calculation here: https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/ecbd6965a0100d9a070110233762593b16023287/pkg/provider/ip/provider.go#L297.

Can you link the ref to the code where this is determined in the resource initialization? I do agree that we should do the resource calculation for `PrivateIPv4` address in a different function

Reply:
Done

Reply:
I'm not sure if we can remove this.
I find the v1.ResourcePods is used a lot in the core Karpenter and it is computed by ENILimitedPods.
If we remove this, the ResourcePods size is larger than the PrivateIPv4Address. I'm not sure if Karpenter will do something wrong in this case.

Please confirm again, if you think it's safe to remove this.

Reply:
I'm not sure if we can remove this.
I find the v1.ResourcePods is used a lot in the core Karpenter and it is computed by ENILimitedPods.
If we remove this, the ResourcePods size is larger than the PrivateIPv4Address. I'm not sure if Karpenter will do something wrong in this case.

Please confirm again, if you think it's safe to remove this.

Reply:
We should validate it, but, in general, we should trust that a resource capacity cannot be exceeded. I would expect that we should see pods (which should have a privateIPv4Address resource injected into them by the vpc-resource-controller mutating webhook) and we should get constrained on the instance type by this resource before pods

Reply:
Removed